tools/lint-docstring

```diff
@@ -120,9 +177,40 @@ def extract_folder_file_structure() -> Dict[str, List[str]]:
     return folders_and_files
 
 
+def extract_docstrings_from_file_overview() -> Dict[str, Dict[str, str]]:
+    """
+    Reads the developer-file-overview.md document and creates
+    a dictionary containing folder name which in turn is a dictionary
+    containing files and their respective descriptons
+    """
+    with open(DEVELOPER_DOC_PATH, "r") as file:
+        doc_data = file.readlines()
+
+    existing_doc_dict: Dict[str, Dict[str, str]] = {}
+
+    # table is present from line 6 to (length - 1)
+    for doc_row in doc_data[TABLE_OFFSET - 1 :]:
+        _, folder, filename, docstring, _ = doc_row.split("|")
+        folder, filename = folder.strip(), filename.strip()
+        if folder:
+            folder_name = folder
+            existing_doc_dict[str(folder_name)] = {}
+        if docstring.strip():
+            if filename:
+                existing_doc_dict[folder_name][filename] = docstring.strip()
```

Comment:
Am I understanding correctly that folder_name is a variable retained through the loop to cope with no folder value when there are files? (eg. the core.py row)
If so, that deserves a comment.

This code combined *could* lead to folder_name being undefined here, if the table was not formatted correctly. It works now as there's a folder at the top of the file.